### ISSUE TITLE
Questions re: user experience

### ISSUE THREAD

**Issue by beaugunderson** [osnr/TabFS#10]
First: this is fantastic work, thank you!

My main question is around the debugging warning that appears in every tab:

![image](https://user-images.githubusercontent.com/61791/103443972-09f6e180-4c19-11eb-8a74-950f8621d186.png)

I'm curious if there are alternate sources for some of the information that wouldn't require the debugger API and thus the debugging warning.

I have a tiny [native host extension](https://github.com/beaugunderson/chrome-tab-server-host) I use to expose my tabs via local HTTP, which I then use Alfred to search, so I can e.g. select a tab by searching URLs and titles when I lose one in a window... It uses the `tabs` permission to access tab metadata with `chrome.windows.getAll({populate: true})`; you get ID, URL, and title from that API.

Would it be useful to use the other extension APIs where possible and have the debugger integration be optional?

Alternately, a good solution for always using `--silent-debugger-extension-api` would be useful, I think (along with documentation of the security concerns if you use other extensions with the `debugger` permission).

**Comment by osnr** [comment]
interesting. I think only stuff [in the `debugger/` subfolder](https://github.com/osnr/TabFS/blob/fef9289e3a7f82cda6319d5f19d5a5f13f3cc44b/extension/background.js#L349) should actually require the debugger and trigger the warning. (i.e. not `title.txt`, `url.txt`, etc).

it's possible that something in your userspace is reaching into `debugger/` (to stat files or whatever) and triggering the debugger (and once it's on, it tends to stay on).

I think if you just delete that whole block I linked, the debugger won't be invoked anymore? (it is already disabled in Firefox, which doesn't have the `chrome.debugger` API)

**Comment by beaugunderson** [comment]
oh interesting! documenting that it shouldn't normally happen would help here, and I think in my case maybe it's Spotlight indexing or something that's enumerating all of the files?

or maybe the initial `grep` or `**/*.txt` enumerated all the debugger-specific files and the debugger remained on for all tabs

**Comment by osnr** [comment]
Yeah. I'll think about how to document that better. I sort of feel like we should have... like... an autogenerated tree / table of contents / listing of all the synthetic files that says what they do, what side effects they have (some pull up the debugger warning, some are slow), etc.

if this is inside the extension, maybe it also has config so you can toggle them off without modifying the source code (although I do want people to feel comfortable modifying the source code, too)
